internal/bsi/bsi.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package bsi.v1;
+option go_package = "github.com/substratusai/substratus/internal/bsi";
+
+service Controller {
+  rpc CreateSignedURL(CreateSignedURLRequest) returns (CreateSignedURLResponse) {}
+}
+
+message CreateSignedURLRequest {
+  string bucketName = 1;
+  string objectPath = 2;
```

Comment:
Originally I didn't have these but realized the controller is calling into the gcs library/service. We should expect the controller to have or build the details of `bucketName` and `objectPath`

Reply:
TODO in the controller integration PR - what bucket is most appropriate to use? Erring toward creating a dedicated bucket for notebook workspace uploads since this is neither a dataset or model.

Reply:
I think we either create an `uploads` bucket, or we create a bucket for each kind and use a subdirectory.

Reply:
notebooks don't currently have a bucket. I think it makes a good case for creating one.

Reply:
renaming `objectPath` to `objectName` which looks more conventional